lms/services/jwt.py

```diff
@@ -115,7 +116,21 @@ def _get_jwk_client(jwk_url: str):
         PyJWKClient maintains a cache of keys it has seen we want to keep
         the clients around with `lru_cache` in case we can reuse that internal cache
         """
-        return jwt.PyJWKClient(jwk_url)
+        return _RequestsPyJWKClient(jwk_url)
+
+
+class _RequestsPyJWKClient(jwt.PyJWKClient):
+    """
+    Version of PyJWKClient which uses requests to gather JWKs.
+
+    Having our own class and using request allows for easier customization.
+    """
+
+    def fetch_data(self):
```

Comment:
Original version at:

https://github.com/jpadilla/pyjwt/blob/a988e1a11e5abb5869dd641f3f4f6a5bb4e70fdf/jwt/jwks_client.py#L20

We could have just add the headers and keep using urllib but using requests feels more "idiomatic" in our codebase.